src/labels_transfer/knn/config.vsh.yaml

```diff
@@ -0,0 +1,103 @@
+functionality:
+  name: knn
+  namespace: "labels_transfer"
+  version: "dev"
+  description: "Performs label transfer from reference to query using KNN classifier"
+  authors:
+    - __merge__: ../../authors/vladimir_shitov.yaml
+      roles: [ author ]
+  argument_groups:
+    - name: Inputs
+      arguments:
+        - name: "--input"
+          alternatives: ["-i"]
+          type: file
+          description: Input h5mu file
+          direction: input
+          required: true
+          info:
+            slots:
+              obsm:
+              - type: "double"
+                name: "X_scanvi"
+                description: "SCANVI or any other embedding on which a classifier is trained. Could have any name, specify it in `query_obsm_key` argument"
+        - name: "--modality"
+          type: string
+          default: "rna"
+          required: false
+        - name: "--reference"
+          type: file
+          description: ".h5ad file with data used to train classifier. If None, HLCA is used"
+          default: "https://zenodo.org/record/6337966/files/HLCA_emb_and_metadata.h5ad"
+          required: false
+          info:
+              slots:
+                obs:
+                - type: "string"
+                  name: "label"
+                  description: "Labels to transfer to query. Could have any name, specify it in `targets` argument"
+                obsm:
+                - type: "double"
+                  name: "X_scanvi"
+                  description: "SCANVI or any other embedding on which a classifier is trained. Could have any name, specify it in `reference_obsm_key` argument"
+        - name: "--targets"
+          type: string
+          default: "ann_level_1,ann_level_2,ann_level_3,ann_level_4,ann_level_5,ann_finest_level"
+          required: false
+          multiple: true
+          multiple_sep: ","
+        - name: "--reference_obsm_key"
+          type: string
+          required: false
+          description: ".obsm key of the data to use as the feature space for training classifier. If None, X is used. It is recommended to use low-dimensional representation (e.g., PCA or scvi embedding)"
+        - name: "--query_obsm_key"
+          type: string
+          required: false
+          description: ".obsm key of the data to use as the feature space for inference of the model. If None, X is used. It is recommended to use low-dimensional representation (e.g., PCA or scvi embedding)"
+    - name: Outputs
+      arguments:
+        - name: "--output"
+          alternatives: ["-o"]
+          type: file
+          description: Output h5mu file.
+          direction: output
+          required: true
+        - name: "--obs_output_suffix"
+          type: string
+          default: "_pred"
+          required: false
+          description: "Suffix to add to each target name in .obs slot to store the predicted classes"
```

Comment:
Why not store this as an obsm? That way you don't need to specify a suffix, but simply a key:

```suggestion

        - name: "--output_obsm_key"
          type: string
          default: "pred"
          required: false
          description: ".obsm key to store the predicted classes" # todo: improve description
```

Also, the format of an argument should be `<arg-file-name>_<slot>_<id>`.